app/goal_routes.py

```diff
@@ -0,0 +1,97 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from .routes_helper import validate_model
+
+goals_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response(jsonify({"details": "Invalid data"}), 400)
```

Comment:
Nice job handling a possible exception if a client sent a request body that didn't have "title" in it. Another way to handle exceptions related to invalid data from the client would be to use the try/except statement. Using try/except is nice because if the attempt to create a Goal object fails, then you can catch the exception and send back an error response to the client.

```python
try:
    new_goal = Goal.from_dict(request_body)
    db.session.add(new_goal)
    db.session.commit()

    response_body = dict(goal = new_goal.to_dict())
    return jsonify(response_body), 201
except KeyError:
        abort(make_response({"details" : "Invalid Data"}))